tensorboard/webapp/app_routing/route_config.ts

```diff
@@ -70,19 +77,25 @@ function getPathFragments(path: string): PathFragment[] {
   });
 }
 
-class RouteConfigMatcher {
-  private readonly pathFragments: PathFragment[];
-  private readonly pathMatchers: PathMatcher[];
-  private readonly redirectionFragments: PathFragment[] | null;
+abstract class RouteConfigMatcher {
+  protected readonly pathFragments: PathFragment[];
+  protected readonly pathMatchers: PathMatcher[];
+  abstract readonly definition: RouteDef;
```

Comment:
Can we drop the `definition` member, since it isn't used by the abstract base class at all, and only 2 of the 3 subclasses?

Reply:
Ditto.